### PR TITLE
Update SEP statuses and README.md

### PR DESCRIPTION
**Pull request by JakeUrban**
This PR updates the statuses of SEPs and improves the readability of the README.

### Changelog

- Creates a table for each status
- Moves SEP-4 from `Final` to `Deprecated`
- Moves SEP-14 from `Draft` to `Final`
- Moves SEP-18 from `Active` to `Final`
- Moves SEP-20 from `Active` to `Final`
- Moves SEP-29 from `Active` to `Final`
- Moves SEP-33 from `Active` to `Final`
- Moves SEP-34 from `Final Comment Period` to `Rejected`
- Moves SEP-35 from `Draft` to `Final`
- Moves SEP-38 from `Draft` to `Active`

### CODE REVIEW

File: ecosystem/sep-0018.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 SEP: 0018
 Title: Data Entry Namespaces
 Author: MisterTicot (@misterticot), Leigh McCulloch (@leighmcculloch)
-Status: Active
+Status: Final
```
Comment (leighmcculloch) [inline ecosystem/sep-0018.md:7]: SEP-18 contains a registry. If we mark it final will the registry still be updatable?

File: ecosystem/sep-0020.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ SEP: 0020
 Title: Self-verification of validator nodes
 Author: Johan Stén <johan@futuretense.io>
 Track: Standard
-Status: Active
+Status: Final
```
Comment (leighmcculloch) [inline ecosystem/sep-0020.md:6]: cc @MonsieurNicolas ?

File: ecosystem/sep-0035.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ SEP: 0035
 Title: Operation IDs
 Author: Scott Fleckenstein (@nullstyle), Isaiah Turner (@Isaiah-Turner), Debnil Sur (@debnil)
 Track: Standard
-Status: Draft
+Status: Final
```
Comment (leighmcculloch) [inline ecosystem/sep-0035.md:8]: 👍

**Review by leighmcculloch — COMMENTED** [review]
Some questions online.

File: ecosystem/sep-0018.md (hunk 1 above)
Comment (JakeUrban) [reply ecosystem/sep-0018.md:7]: I see two options,

1. SEPs in 'Final' cannot be updated in any form
2. SEPs in 'Final' can have patch-version updates

Personally I think the first option is the way to go, but that means we'd need to put the registry somewhere else.

**Pull request closed without merging**